Tools/scripts/freeze_modules.py

```diff
@@ -21,20 +23,48 @@
 MODULES_DIR = os.path.join(ROOT_DIR, 'Python/frozen_modules')
 TOOL = os.path.join(ROOT_DIR, 'Programs', '_freeze_module')
 
+MANIFEST = os.path.join(MODULES_DIR, 'MANIFEST')
 FROZEN_FILE = os.path.join(ROOT_DIR, 'Python', 'frozen.c')
+GITIGNORE = os.path.join(ROOT_DIR, '.gitignore')
 MAKEFILE = os.path.join(ROOT_DIR, 'Makefile.pre.in')
 PCBUILD_PROJECT = os.path.join(ROOT_DIR, 'PCbuild', '_freeze_module.vcxproj')
 PCBUILD_FILTERS = os.path.join(ROOT_DIR, 'PCbuild', '_freeze_module.vcxproj.filters')
+TEST_CTYPES = os.path.join(STDLIB_DIR, 'ctypes', 'test', 'test_values.py')
+
+
+OS_PATH = 'ntpath' if os.name == 'nt' else 'posixpath'
 
 # These are modules that get frozen.
 FROZEN = [
     # See parse_frozen_spec() for the format.
     # In cases where the frozenid is duplicated, the first one is re-used.
-    ('importlib', [
+    ('import system', [
+        # These frozen modules are necessary for bootstrapping
+        # the import system.
         'importlib._bootstrap : _frozen_importlib',
         'importlib._bootstrap_external : _frozen_importlib_external',
+        # This module is important because some Python builds rely
+        # on a builtin zip file instead of a filesystem.
         'zipimport',
         ]),
+    ('stdlib', [
+        # without site (python -S)
+        'abc',
+        'codecs',
+        '<encodings.*>',
+        'io',
+        # with site
+        '_collections_abc',
+        '_sitebuiltins',
+        'genericpath',
+        OS_PATH,
+        # We must explicitly mark os.path as a frozen module
+        # even though it will never be imported.
+        f'{OS_PATH} : os.path',
```

Comment:
> But why does the comment says it will never be imported? Because import os automatically sets os.path?

That's right.

> But then why does it need to be in the table? What kind of code fails without this?

Some methods of the importer for frozen modules check to make sure the requested module is a frozen module by calling `_imp.is_frozen()` on the module name.  So if "os.path" isn't in the array of frozen modules, the `is_frozen()` call will return False, even though the aliased module (e.g. posixpath) actually is.

The simplest solution was to add `os.path` to the list of frozen modules.  An added benefit is that we don't have to add any special case logic in the importlib code.